### PR TITLE
Add the ability to create new lists

### PR DESCRIPTION
**Pull request by samstradling**
When the plus icon is pressed on the list selector a new list is created. At the moment it is not possible to swap between the lists.

### CODE REVIEW

File: src/hooks/useAppState/reducer.js [hunk 1]
```diff
@@ -15,10 +15,11 @@ export const initialState = { lists: [], index: 0 };
 const reducer = (state, { type, data }) => {
   switch (type) {
     case types.NEW_LIST: {
-      const { key, name } = data;
+      const { key, name, id } = data;
       return {
         ...state,
-        lists: [...state.lists, { key, name, items: [] }],
+        lists: [...state.lists, { key, name, id, items: [] }],
+        index: state.lists.length,
```
Comment (samstradling) [inline src/hooks/useAppState/reducer.js:22]: This means that the new list is selected after it is created which I presume is the behaviour we're after

File: src/components/ListSelector/ListSelector.js [hunk 2]
```diff
@@ -22,6 +22,7 @@ const ListSelector = ({ lists, selectedIndex }) => (
         <FontAwesomeIcon
           icon={faPlusCircle}
           className={styles.addListIcon}
+          onClick={() => actions.newList(`Shopping List ${lists.length + 1}`)}
```
Comment (samstradling) [inline src/components/ListSelector/ListSelector.js:25]: The list name will be `Shopping List X` where `X` increments for each new list.

File: src/hooks/useAppState/middleware.js [hunk 3]
```diff
@@ -60,6 +60,13 @@ const applyMiddleware = (dispatch, getState) => (action) => {
       break;
     }
 
+    case types.NEW_LIST: {
+      const { key, name } = data;
+      createList(name)
+        .then((id) => dispatch({ type: types.STORE_LIST_ID, data: { key, id } }));
+      break;
+    }
```
Comment (samstradling) [inline src/hooks/useAppState/middleware.js:68]: When a new list is created we call the API to create the list with that name, this gives us an ID which we store so we can changes later to that list (e.g. changing its name)

File: src/hooks/useAppState/reducer.js (hunk 1 above)
Comment (Joseph-Cashmore) [reply src/hooks/useAppState/reducer.js:22]: Yes definitely

**Review by Joseph-Cashmore — APPROVED** [review]

**Pull request merged**